index.js

```diff
@@ -11,11 +13,38 @@ server.get('/payments', (req, res) => {
     res.status(200).send(payments);
 });
 
-
+function validatePaymentDataMiddleWare(req, res, next) {
+    const payment = {...req.body};
+    const currentDate = new Date().toLocaleDateString('fr-CA');
+    
+    if(!(payment.date >= currentDate &&
+        currencies.includes(payment.currency) &&
+        Number.isFinite(payment.amount) && 
+        Number.isFinite(payment.exchangeRate) &&
+        payment.description &&
+        payment.description !== '')) {
+            res.status(400).send('Invalid Data');
+            return;
```

Comment:
hi 
we should not have return here. 
please refer to this documentation: https://expressjs.com/en/guide/writing-middleware.html 
we just send(BAD_REQUEST) status
and the always finish our validation handler with calling `next()`. if we do `return`  here, then `next()` doesn't get called